src/mscorlib/src/System/Globalization/TextInfo.cs

```diff
@@ -482,6 +482,15 @@ public unsafe String ToTitleCase(String str)
                 charType = CharUnicodeInfo.InternalGetUnicodeCategory(str, i, out charLen);
                 if (Char.CheckLetter(charType))
                 {
+                    // Special case to check for Dutch specific titlecasing with
+                    // "IJ" characters at the beginning of word
+                    if (IsDutchCulture(str) && IsIjAtCurrentPosition(str, i))
+                    {
+                        // Increment the charLen to treat the "IJ" as a surrogate
+                        // pair and capitalize both letters
+                        charLen++;
+                    }
+
```

Comment:
It is better we optimize this case as AddTitlecaseLetter call allocate extra string. we can do the following:

if (isDutchCulture && i < str.Lenth - 1 && (str[i] == 'i" || str[i] == 'I") && (str[i+1] == 'j' || str[i+1] == 'J') )
{
    result.Append("IJ");
    i+= 2;
}